src/main/java/backend/resource/MultiModel.java

```diff
@@ -207,9 +207,14 @@ private void preprocessUpdatedIssues(List<Model> newModels) {
                     // TODO O(n^2), optimise by preprocessing into a map or sorting
                     if (!existingModel.getIssueById(i).equals(model.getIssueById(i))) {
                         assert model.getIssueById(i).isPresent();
-                        // It's no longer currently read, but it retains its updated time.
-                        // No changes to preferences.
-                        model.getIssueById(i).get().setIsCurrentlyRead(false);
+                        /** If the issue was marked read after the last update, but update came in later
+                         * due to network latency, ensure the issue retains its read state
+                        **/
+                        if (model.getIssueById(i).get().getUpdatedAt().isAfter(model.getIssueById(i).get().getMarkedReadAt().get()) 
```

Comment:
That's a monster conditional... you should factor it out into a predicate. Binding `model.getIssueById(i)` will probably help.

Also you need to handle the case where `markedReadAt` is not present, otherwise it can fail at runtime like a dereferenced null pointer.